src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java

```diff
@@ -100,51 +102,22 @@ public void runLocal(String queryFile, String outputPath) throws IOException {
             System.out.println(sb);
             return;
         }
+
         if (result.isRDD() && outputPath != null) {
             JavaRDD<Item> rdd = result.getRDD(dynamicContext);
             JavaRDD<String> output = rdd.map(o -> o.serialize());
             output.saveAsTextFile(outputPath);
         } else {
             String output = runIterators(result, dynamicContext);
             if (outputPath != null) {
+                File outputFile = new File(outputPath);
                 List<String> lines = Arrays.asList(output);
                 org.apache.commons.io.FileUtils.writeLines(outputFile, "UTF-8", lines);
```

Comment:
The plan is to also use the hadoop filesystem for writing back small output, similar to everything else, but I have not gotten to it yet.